### DISCUSSION TITLE
Extracting count matrices for individual samples after integration

### DISCUSSION THREAD

**Discussion by VPetukhov** [satijalab/seurat#4125] [Q&A] [answered]
In Kharchenko lab we're developing a package for case-control analysis of scRNA-seq experiments, which we want to integrate with Seurat. And for many analyses we need to extract count matrices for each of the samples from the integrated Seurat object. Is it possible to do without having a vector with sample labels in advance?

@mojaveazure , could you please help us with that?

**Comment by timoast** [comment, marked as answer]
Hi @VPetukhov, for efficiency we don't store the original counts in the integrated assay (because we'd be then be storing them twice in the Seurat object). The original counts can be found in the RNA assay (or whatever the name of the original assay was that you integrated). You can access this using:

```r
counts <- GetAssayData(integrated_object, assay = "RNA", slot = "counts")
```

If you want to know which cell was from which object originally, you'd need some additional metadata that describes this.
